src/tequila/simulators/simulator_cirq.py

```diff
@@ -138,6 +163,38 @@ def add_measurement(self, gate, circuit, *args, **kwargs):
     def make_qubit_map(self, qubits) -> typing.Dict[numbers.Integral, cirq.LineQubit]:
         return {q: cirq.LineQubit(i) for i,q in enumerate(qubits)}
 
+    def build_device_circuit(self,device):
+        c = self.circuit
+        if isinstance(device,str):
```

Comment:
Avoid explicit case handling at every level. Devices should be assigned correctly at that stage